integration/src/test/java/org/mule/test/routing/AsyncTestCase.java

```diff
@@ -119,6 +121,41 @@ public void withinSubflowDoesntUseFlowMaxConcurrency() throws Exception {
     latch.countDown();
   }
 
+  @Test
+  @Description("Assert that async blocks run outside of the transaction from the caller flow")
+  public void withSourceTx() throws Exception {
+    terminationLatch = new CountDownLatch(0);
+
+    final Startable withSourceTx =
+        (Startable) (locator.find(Location.builderFromStringRepresentation("with-source-tx").build())).get();
+    withSourceTx.start();
+
+    assertThat(queueHandler.read("asyncDispatched", 1000), not(nullValue()));
+    assertThat(queueHandler.read("asyncRunning", 1000), not(nullValue()));
+  }
+
+  @Test
+  @Description("Assert that async blocks run outside of the transaction from the caller flow")
+  public void withTryTx() throws Exception {
+    CountDownLatch latch = new CountDownLatch(1);
+    runFlows("with-try-tx", latch);
```

Comment:
Why don't these need starting?

Reply:
the flowRunner already starts the flow if needed

Reply:
https://github.com/mulesoft/mule/blob/mule-4.x/tests/component-plugin/src/main/java/org/mule/functional/api/flow/FlowRunner.java#L76